backend/service/martini/email.go

```diff
@@ -0,0 +1,40 @@
+package martini
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/ekspand/trusty/pkg/email"
+)
+
+// renderEmailTemplate used to render email template with given data
+func renderEmailTemplate(html string, data interface{}) (string, error) {
+	tmpl, err := template.New("email").Parse(html)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse template, err=[%v]", err)
+	}
+
+	var buf bytes.Buffer
+
+	if err := tmpl.Execute(&buf, data); err != nil {
+		return "", fmt.Errorf("failed to render email template, err=[%v]", err)
+
+	}
+	return buf.String(), nil
+}
+
+// sendEmail used to send email with Mailgun provider
+func (s *Service) sendEmail(toEmail string, subject, string, htmlTemplate string, data interface{}) error {
+	body, err := renderEmailTemplate(htmlTemplate, data)
+
+	if err != nil {
+		return fmt.Errorf("failed to render email template, err=[%v]", err)
```

Comment:
replace all `fmt.Errorf("failed to render email template, err=[%v]", err)` with
`errors.Annotatef(err, "failed to xxx")`